### PR TITLE
Fix Reduce layer for cosnt inputs

### PR DESCRIPTION
**Pull request by Abdurrahheem**
### Pull Request Readiness Checklist

This PR adds support for const inputs for reducing the layer. Particularly, it fixes the following case. The test model and data are located in [1194](https://github.com/opencv/opencv_extra/pull/1194)

image

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Review by fengyuentau — COMMENTED** [review]
What is n2 and n4 in the graph?

File: modules/dnn/src/onnx/onnx_importer.cpp [hunk 1]
```diff
@@ -1209,6 +1209,15 @@ void ONNXImporter::parseReduce(LayerParams& layerParams, const opencv_onnx::Node
         for (int i = 0; i < num_axes; ++i)
             axes[i] = mat_axes.at<int>(i);
         layerParams.set("axes", DictValue::arrayInt(&axes[0], num_axes));
+        if (constBlobs.find(node_proto.input(0)) != constBlobs.end()){
+            std::vector<Mat> inputs, output;
+            inputs.push_back(getBlob(node_proto, 0));
+            layerParams.type = "Reduce";
```
Comment (fengyuentau) [inline modules/dnn/src/onnx/onnx_importer.cpp:1215]: Move this line to the beginning of `parseReduce`. Then line 1223 can be removed.

**Comment by Abdurrahheem** [comment]
> What is n2 and n4 in the graph?

`n2` - pow operation
`n4` - reduce operation

Do not know why proper names do not show up in the graph constructed using onnxscript

**Comment by fengyuentau** [comment]
> > What is n2 and n4 in the graph?
> 
> `n2` - pow operation `n4` - reduce operation
> 
> Do not know why proper names do not show up in the graph constructed using onnxscript

I donwloaded the model you genereated and it looks good with netron on my side:

image

File: modules/dnn/src/onnx/onnx_importer.cpp (hunk 1 above)
Comment (Abdurrahheem) [reply modules/dnn/src/onnx/onnx_importer.cpp]: Done

**Comment by Abdurrahheem** [comment]
I'd like to add a general note on why this PR was needed. 

The reduce layer would not know the parser graph where inputs are constants. This PR extends the functionality of the parser in dealing with such cases. The graph of the test model has an accessive branch, `x`. This branch is useless and only was introduced to obey the API of the model generation [script](https://github.com/opencv/opencv_extra/blob/02f795794515613b94e6ec473f1e7c3ae631c68d/testdata/dnn/onnx/generate_onnx_models_with_onnxscript.py#L38-L42), which requires input to exist (as there is no input if all inputs are constants).

**Review by dkurt — APPROVED** [review]

**Pull request merged**